### PR TITLE
implementation of adding fiches to the database

### PR DESCRIPTION
**Pull request by paweldowgiallo84**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task goal
The task goal was to save a new flashcard to the database when adding one.

### Requirements
1. Fetch API must be used.
2. The following endpoint must be used:
```
curl -X POST \
  https://training.nerdbord.io/api/v1/fischkapp/flashcards \
  -H 'Authorization: "secret_token" \
  -H 'Content-Type: application/json' \
  -d '{
    "front": "What is life?",
    "back": "I have no idea.",
}'
```

### Code Review

#### src/App.tsx
The changes in this file seem to align with the task requirements. The `FISCHKAPP_URL` and `URL_AUTH_TOKEN` constants have been exported for use in other components, which is necessary for making the POST request to the specified endpoint with the required headers.

The `addNewCard` function has been removed, which is consistent with the task requirements as the new card should now be added via a POST request to the server, not by updating the local state.

#### src/components/AddCard/AppAddCard.tsx
The changes in this file also align with the task requirements. The `fetch` API is used to make a POST request to the specified endpoint with the required headers and body. The body of the request is correctly formatted as a JSON string.

The function `addCardData` checks if the question and answer fields are not empty before making the request, which is a good practice to prevent sending invalid data to the server. If the request is successful, the input fields are cleared and the card adding form is closed, which is a good user experience.

However, there is a minor issue with error handling. If the request fails, an error message is logged to the console, but it would be better to also display an error message to the user. This could be done by setting the `errorMsg` state to a relevant error message in the catch block of the fetch promise.

Here is an example of how you could improve the error handling:

```diff
 .catch(Error => {
   console.error('Error: ', Error)
+  setErrorMsg('Failed to add card. Please try again.')
 })
```

In conclusion, the code changes cover the task requirements. The new flashcard is saved to the database using the Fetch API and the specified endpoint. The code is generally well-written and follows good practices. The only suggestion for improvement is to enhance the error handling by displaying an error message to the user when the request fails.
